main.tf

```diff
@@ -0,0 +1,314 @@
+  sink_output = {
+    resource_name = "${local.is_project_level ? var.project : local.is_folder_level ? var.folder : local.is_org_level ? var.org_id : ""}"
+    resource      = "${local.is_project_level ? element(concat(google_logging_project_sink.sink.*.id, list("")), 0) : local.is_folder_level ? element(concat(google_logging_folder_sink.sink.*.id, list("")), 0) : local.is_org_level ? element(concat(google_logging_organization_sink.sink.*.id, list("")), 0) : local.is_billing_level ? element(concat(google_logging_billing_account_sink.sink.*.id, list("")), 0) : ""}"
+    writer        = "${local.is_project_level ? element(concat(google_logging_project_sink.sink.*.writer_identity, list("")), 0) : local.is_folder_level ? element(concat(google_logging_folder_sink.sink.*.writer_identity, list("")), 0) : local.is_org_level ? element(concat(google_logging_organization_sink.sink.*.writer_identity, list("")), 0) : local.is_billing_level ? element(concat(google_logging_billing_account_sink.sink.*.writer_identity, list("")), 0) : ""}"
+    console_link  = "${local.is_project_level ? "https://console.cloud.google.com/logs/exports?project=${var.project}" : ""}"
+    name          = "${var.name}"
+    filter        = "${var.filter}"
+  }
+
+  # Sink destination output data
+  destination_output = {
+    resource     = "${local.is_pubsub ? element(concat(google_pubsub_topic.topic.*.id, list("")), 0) : local.is_bigquery ? element(concat(google_bigquery_dataset.dataset.*.id, list("")), 0) : local.is_storage ? element(concat(google_storage_bucket.bucket.*.id, list("")), 0) : ""}"
+    self_link    = "${local.destination_uri}"
+    console_link = "${local.destination_console_link}"
+    type         = "${local.destination_type}"
+    project      = "${local.is_pubsub ? element(concat(google_pubsub_topic.topic.*.project, list("")), 0) : local.is_bigquery ? element(concat(google_bigquery_dataset.dataset.*.project, list("")), 0) : local.is_storage ? element(concat(google_storage_bucket.bucket.*.project, list("")), 0) : ""}"
+  }
+}
+
+#---------------------#
+# Logsink destination #
+#---------------------#
+
+# Enable destination API
+resource "null_resource" "enable_destination_api" {
+  provisioner "local-exec" {
+    command = "sh ${path.module}/scripts/enable-api.sh ${local.destination_api} ${local.destination_project}"
+  }
+}
+
+# Pub/Sub topic
+resource "google_pubsub_topic" "topic" {
+  count   = "${local.destination_type == "pubsub" ? 1 : 0}"
+  name    = "${local.destination_name}"
+  project = "${local.destination_project}"
+
+  depends_on = [
+    "null_resource.enable_destination_api",
+  ]
+}
+
+# BigQuery dataset
+resource "google_bigquery_dataset" "dataset" {
+  count      = "${local.destination_type == "bigquery" ? 1 : 0}"
+  dataset_id = "${local.destination_name}"
+  project    = "${local.destination_project}"
+
+  # Delete all tables in dataset on destroy.
+  # This is required because a dataset cannot be deleted if it contains data
```

Comment:
Is there no way to force this? If not let's open an issue on the Terraform provider to request adding a flag for handling that.

Reply:
Yes, but please open an issue on this repo to track implementing that once upstream gets fixed. Thanks.